test/libsolidity/semanticTests/events/event_indexed_function.sol

```diff
@@ -0,0 +1,11 @@
+contract C {
+    event Test(function() external indexed);
+    function f() public {
+        emit Test(this.f);
+    }
+}
+// ====
+// compileViaYul: also
+// ----
+// f() ->
+// ~ emit: #0x16e407cc31a66b716200af7eadb4155035b14f295af52acbc4356b464d649b71, #0x0fdd67305928fcac8d213d1e47bfa6165cd0b87b26121ff00000000000000000
```

Comment:
Why is this event missing a signature?

Reply:
I guess it's related to the function type in the signature. I will check that.

Reply:
I use the contract abi to reconstruct the signature. I was wondering what the difference between `internalType` and `type` is. Initially I used `internalType`, where it be `function () external` in this case. After checking the code generator I saw that only `function` is used here. So I just change the signature function to use `type`. 

However, what I found strange is that the signature of the event `Test(function() external indexed);` is `Test(function)`. But an event `Test(function(uint256) external indexed);` will also be `Test(function)`. I would have expect something like `Test(function(uint256)`. Is this the desired behaviour?

Reply:
See https://github.com/ethereum/solidity/blob/624b482f50e110b6e7df74f73d125502c8d1cbf0/test/libsolidity/semanticTests/events/event_indexed_function2.sol

Reply:
That's a good question. I'd expect `Test(function(uint256))` too but apparently all external function types have the same ABI encoding regardless of their arguments.

I could not find any mention of function types in the docs for how a selector is constructed (we should really document that; #11184) but I did an experiment with 0.8.4 and it looks like you cannot overload on function types:

```solidity
contract C {
    struct S {uint a; uint b;}

    function foo(function() external) public {}
    function foo(function(uint) external) public {}
    function foo(function(S memory) external) public {}
}
```
```
Error (9914): Function overload clash during conversion to external types for arguments.
```
I think that this means that they have the same signature.